### ISSUE TITLE
Question about get class agnostic masks using mask3d

### ISSUE THREAD

**Issue by lziiid** [aminebdj/OpenYOLO3D#20]
Hi, thanks for your open-sourced work! My question is about this function:
https://github.com/aminebdj/OpenYOLO3D/blob/400abe8ba2faa282218ecdf695f1810d16a8c5fa/models/Mask3D/mask3d/__init__.py#L207-L236

I am confused about the function, especially about line 221. My questions are:
1. The logits shape is [num_masks, num_classes] ([150, 201] in scannet200 cases). So what's the meaning of the`scores_per_query, topk_indices = logits.flatten(0, 1).topk(num_queries, sorted=True)` ? From my understanding, comparing logits between different masks is meaningless, since logits(without softmax) are not possibilities.
2. Without considering question 1, `topk_indices` means the topk highest scores in logits. So why use 200 in `topk_indices = topk_indices // 200`? 201 is the second dimension of logits, not the first. I think the `topk_indices` is meaningless.
3. So I don't understand `masks = masks[:, topk_indices]` too... 
4. From my perspective, I think the correct code should be something like:
```python
for i in range(len(logits)):
        p_labels = torch.softmax(logits[i], dim=-1)
        p_masks = torch.sigmoid(masks[:, i])
        l = torch.argmax(p_labels, dim=-1)
        c_label = torch.max(p_labels)
        m = p_masks > 0.5
        c_m = p_masks[m].sum() / (m.sum() + 1e-8)
        c = c_label * c_m
        if l < 200 and c > confidence_threshold:
            labels.append(l.item())
            confidences.append(c.item())
            masks_binary.append(
                m[inverse_map])  # mapping the mask back to the original point cloud
```
 in https://github.com/cvg/Mask3D/blob/7ae86b759a1d6838945a5c2e1204effdae1b9d07/mask3d/__init__.py#L165-L177
which I find in the original Mask3D repository.

I would be extremely grateful if you could solve this difficult problem for me.
